src/slashtag-server.js

```diff
@@ -1,45 +1,34 @@
-import config from 'config'
-import { SDK } from '@synonymdev/slashtags-sdk'
-import logger from './logger.js'
-import SeedingProtocol from './seeding-protocol.js'
-
-export default class SlashServer {
-    constructor(seeder) {
-        this.seedingProtocol = null
-        this.seeder = seeder
-    }
-
-    async start() {
-        // Pick a primary key, so all our slashtags are the same each time
-        const keyStr = config.get('slashtags.primaryKey')
-        if (keyStr.length !== 64) {
-            throw new Error('slashtags.primaryKey should be defined in the config (expecting 64 character hex string)')
-        }
-
-        // as a buffer
-        const primaryKey = Buffer.from(keyStr, 'hex')
-
-        // setup the SDK
-        const options = {
-            persist: false,
-            primaryKey
-        }
-
-        // start the slashtags SDK
-        const sdk = new SDK(options)
-        await sdk.ready()
-
-        // Generate the servers slashtag (as we use a common name and primary key, this should be the same each time)
-        const st = sdk.slashtag('Seeding Server Slashtag');
-        logger.info(`Seeding Server Slashtag - ${st.url}`)
-
-        // Get the seeding protocol to add our hooks
-        this.seedingProtocol = new SeedingProtocol(st)
+const logger = require('./logger.js')
+const SeederRPC = require('./seeding-protocol.js')
+const Seeder = require('./seeder.js')
+
+/**
+ * Transport over HyperDHT connections as an alternative to the HTTP server.
+ */
+class SlashServer {
```

Comment:
@Nuhvi We need to develop a generic Slashtags RPC server abstraction. Fastify for Slashtags that can be used everywhere that we need to have a RPC server.

Reply:
Happy to use this for now and then when the generic RPC server abstraction is done, we migrate this

Reply:
We do have a generic [RPC](https://www.npmjs.com/package/@synonymdev/slashtags-rpc) that we extend everywhere including in this repository [here](https://github.com/synonymdev/slashtag-seeding-server/blob/master/src/seeding-protocol.js#LL6C46-L6C58)